drgn/helpers/linux/cgroup.py

```diff
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+"""
+Cgroup
+------
+
+The ``drgn.helpers.linux.cgroup`` module provides helpers for working with
+cgroup interface in :linux:`include/linux/cgroup.h`. Only cgroup v2 is
+supported.
+"""
+
+import os
+
+from drgn import Object, container_of
+
+
+__all__ = [
+        'cgroup_name',
+        'cgroup_parent',
+        'cgroup_path',
+        'sock_cgroup',
+]
+
+
+def sock_cgroup(skcd):
+    """
+    .. c:function:: struct cgroup *sock_cgroup(struct sock_cgroup_data *skcd)
+
+    Get cgroup of a socket by ``struct sock_cgroup_data`` member embedded to
+    every ``struct sock``.
+
+
+    :return: ``struct cgroup`` object.
+    """
+    return Object(skcd.prog_, 'struct cgroup', address=skcd.val)
+
+
+def cgroup_parent(cgrp):
+    """
+    .. c:function:: struct cgroup *cgroup_parent(struct cgroup *cgrp)
+
+    Get parent of cgroup if it exists.
+
+    :return: ``struct cgroup`` object or None.
+    """
+    parent_css = cgrp.self.parent
+    if parent_css:
+        return container_of(parent_css, 'struct cgroup', 'self')
+    return None
+
+
+def cgroup_name(cgrp):
+    """
+    .. c:function:: char *cgroup_name(struct cgroup *cgrp)
+
+    Get cgroup name.
+
+    :rtype: bytes
+    """
+    kn = cgrp.kn
+    return kn.name.string_() if kn.parent else b'/'
+
+
+def cgroup_path(cgrp):
+    """
+    .. c:function:: char *cgroup_path(struct cgroup *cgrp)
+
+    Get cgroup path.
+
+    :rtype: bytes
+    """
+    names = []
+    while cgrp is not None and cgrp.level >= 0:
+        names.append(cgroup_name(cgrp))
+        cgrp = cgroup_parent(cgrp)
+    names.reverse()
+    return os.path.join(b'', *names)
```

Comment:
This is really pedantic, but I'd prefer `b'/'.join(names)` over `os.path.join()`, because we're joining a Linux path, not a path on whatever operating system drgn is running on. And who knows, maybe someone will port drgn to Windows and use it to debug a Linux core dump :)

Reply:
Yeah, I saw it in other helpers, but wasn't sure "why" :) OK, makes sense, I'll change it -- not a big deal.